packages/dusk/package.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+#
+from spack import *
+import sys
+
+class Dusk(PythonPackage):
+    """A minimal and lightweight front-end for dawn."""
+
+    homepage = 'https://github.com/dawn-ico/dusk'
+    git      = 'https://github.com/dawn-ico/dusk'
+    maintainers = ['BenWeber42']
+
+    version('master', branch='master')
+    depends_on('python@3.8.0', type=('build', 'run'))
+
+    depends_on('dawn4py')
+    depends_on('py-setuptools', type='build')
+    depends_on('py-pip', type='build')
```

Comment:
```suggestion
    # We only want Python 3.8.X, apparently that's the way to specify it...
    # https://spack.readthedocs.io/en/latest/packaging_guide.html#version-ranges
    extends('python@3.8.0:3.8.999')

    depends_on('dawn4py', type='run')
    depends_on('py-setuptools', type='build')

    # will test that dusk is importable after the install
    import_modules = ['dusk']
```
While the current package works:
* I think `extends` is better in this case.
* We only need `dawn4py` at runtime
* We don't need `py-pip`, because dusk only depends on `dawn4py` by github url (not pip)
* Checking whether `dusk` is importable after the install is a cheap and simple test (we could add dusk's tests as well, but that makes it more complicated and I think spack is still consolidating how tests can be run after installation)

What do you think?

Reply:
Applying these changes to check if they work also with icondusk-e2e's CI. 
Only objection is the dependency on `dawn4py` via github url: in general we want to configure at setup time which version of the dependency to use (possibly one that was already installed). In the example of icondusk-e2e's CI we select (through the launch jenkins... comment) which dawn and which dusk.
So in dusk's `setup.cfg`, does the following line select the dawn4py which is already present? Or will it fetch the latest master from github in any case?
```
install_requires =
  dawn4py@git+https://github.com/MeteoSwiss-APN/dawn.git#subdirectory=dawn
```
I could go ahead and test it, but maybe you already know.

Reply:
Good question. I think with `pip` in a virtualenv it won't install `dawn4py` if it's already installed.
But it probably depends on what `setuptools` does with this line (not `pip`).

So I tested this by running `python setup.py build` (works fine without `dawn4py`) and `python setup.py install` (fails without `dawn4py`). If `dawn4py` however is already installed in the virtualenv, both will succeed.

I guess you are right then, `dawn4py` is a build dependency (technically _install_, but this isn't a valid dependency type in spack, so it's build).